### PR TITLE
feat(StatusQ.Components): Adding StatusToastMessage

### LINKED ISSUE #521: Implement `StatusToastMessage` component
figma: https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=2090%3A252386
figma: https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=1838%3A360271
figma: https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=1939%3A218046

Screen Shot 2022-01-12 at 3 19 20 PM

### PR DESCRIPTION
**Pull request by alexandraB99**
Added StatusToastMessage component, an example page
as well as a demonstration in chat view

Closes #521

https://user-images.githubusercontent.com/31625338/152238220-b533f300-8da8-49a0-a262-125ed7b990da.mov

### CODE REVIEW

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [inline src/StatusQ/Components/StatusToastMessage.qml:38]: Since this is not a select component, we shouldn't use that palette color.
Let's rather introduce a `statusToastMessage` namespace for that.

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [inline src/StatusQ/Components/StatusToastMessage.qml:85]: Not sure what this is about, but if this is some sort of default color for this component, it should live in n`Theme`.

**Review by PascalPrecht — COMMENTED** [review]
Very nice and clean @alexandraB99 !
Just left some comments related to colors.

Will review again once the "don't" merge label has been removed.

**Comment by alexandraB99** [comment]
@PascalPrecht took care of the comments, you can have a look now. I added also a page in the API documentation. That said I am not sure how far we should go with this task, since it's only about creating the component. The mechanism to show/hide notifications lacks specification and I assume it will also depend on how the backend will be designed (if it's not there already). So does it make sense to implement also the mechanism and show it in demoapp in addition to the component page, or it's sufficient to just demonstrate the component in API documentation for this task?

**Comment by alexandraB99** [comment]
after discussion with @iurimatias, this PR fulfils the task. The open/close mechanism should be tackled in a different task.

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [inline src/StatusQ/Components/StatusToastMessage.qml:150]: Judging by the designs, this text should always have the link color, not only when hovered.

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [inline src/StatusQ/Components/StatusToastMessage.qml:20]: I think it'd be nicer if we used `StatusIconSettings` here instead and then give it some decent defaults. This way, we stay consistent with other components that render icons.

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [inline src/StatusQ/Components/StatusToastMessage.qml:21]: I'd expect that there's toast messages that are neither "success" nor "loading" but probably "failure".

How about we rather introduce an enum type, similar to many other components in StatusQ that can have these different states?

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [inline src/StatusQ/Components/StatusToastMessage.qml:157]: This should rather be a 

```qml
        StatusFlatRoundButton {
            type: StatusFlatRoundButton.Type.Secondary
            ...
        }
```

**Review by PascalPrecht — CHANGES_REQUESTED** [review]
> you can have a look now. I added also a page in the API documentation.

Wonderful @alexandraB99 !

I've found a few more things that need adjustment, but nothing major.

Re: adding it to demo app: 
Generally, API documentation page is a must. Demo app scenario is a nice to have.

Once the outstanding things are addressed, this is gtg!

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (jrainville) [inline src/StatusQ/Components/StatusToastMessage.qml:14]: Should the anchors be handled by the component that uses it instead?

**Review by jrainville — COMMENTED** [review]
Looks pretty good to me. I'm not used to reviewing StatusQ components, so I'll trust Pascal's opinion on the matter

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (alexandraB99) [reply src/StatusQ/Components/StatusToastMessage.qml:150]: this is not clear to me as it's not really specified, only one state is shown so I can't judge. @[John-44](https://github.com/John-44) could you maybe clarify?

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (alexandraB99) [reply src/StatusQ/Components/StatusToastMessage.qml:14]: this is to handle the animation for show/hide. Atm I don't have a view of how that should be working since very limited info is available on Figma. This should be re-visited when we know how toast messages should work exactly

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (alexandraB99) [reply src/StatusQ/Components/StatusToastMessage.qml:20]: I thought about this, but since is nothing specified yet, I don't know whether it will be only the icon name needed, more or nothing (maybe we have fixed icons). I'd avoid adding complexity if not needed. nevertheless this is something we can always add later when more details are available.

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [reply src/StatusQ/Components/StatusToastMessage.qml:150]: If we keep it like it is now, you basically don't have an indicator that this text is clickable in the first place and it looks like the other case where there's indeed only info text. 

I assume that's why the "View on Etherscan" text has the primary color (blue) by default. I'd also then re-add the text-decoration: underline on hover to get the hover indicator back.

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [reply src/StatusQ/Components/StatusToastMessage.qml:20]: I understand your reasoning. Just be aware that, if we do need to change it afterwards, it will introduce a breaking change (which I'd avoid as much as possible) unless we keep the `iconName` property around.

I'll let this be your call.

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [reply src/StatusQ/Components/StatusToastMessage.qml:21]: Same as with the icon settings: We can add it later, but it would require us to change the API so this will be a breaking change.

> I thought about this, but since is nothing specified yet I don't know whether it will be only 2 types or more

We already have "failure" toast messages in Status Desktop today (for failing transactions). The new design just seems to be incomplete on that regard. I guess we want to preserve existing functionality.

**Comment by alexandraB99** [comment]
@PascalPrecht it's not that I disagree, but that I can't decide whether type and icon settings are needed or not since a lot of details are still missing. Anyway, I took care of all comments so you could have a second look. When we have more details we can update accordingly.

File: src/StatusQ/Components/StatusToastMessage.qml
Comment (PascalPrecht) [inline src/StatusQ/Components/StatusToastMessage.qml:37]: If you went for `StatusIconSettings`, this should probably now be `root.icon.name = iconName`

Comment (alexandraB99) [reply src/StatusQ/Components/StatusToastMessage.qml]: you are right, updated!

**Comment by PascalPrecht** [comment]
This has landed as https://github.com/status-im/StatusQ/commit/7b290ddd56bb228c5341147699246a0fe7680b70

@alexandraB99 I made a few smaller changes:

1. The close button didn't have the correct variation (needed to be a secondary type variant)
2. On hover, the close button size was way too big, so I've updated as well to match what we have in the rest of the design system

Here's a screenshot:
Screenshot 2022-02-08 at 09 36 24

Massive work! Happy to see this has landed :)

**Pull request closed without merging**
